### ISSUE TITLE
IDE0120 does not work in various cases (Where+Count)

### ISSUE THREAD

**Issue by arphox** [dotnet/roslyn#71293]
**Version Used**: I don't know how to check it (maybe consider adding a section about it to [CONTRIBUTING.md](https://github.com/dotnet/roslyn/blob/main/CONTRIBUTING.md)?).

IDE0120 does not work in various cases (Where+Count), I made some example for that:

```cs
private static void Test()
{
	IEnumerable<string> words = new List<string> { "hello", "world", "!" };
	int cnt1 = words.Where(x => x.Equals("hello")).Count(); // IDE0120 reported

	// Examples where it is not reported:
	int cnt2 = new List<string>().Where(x => x.Equals("hello")).Count();
	int cnt3 = s_wordsField.Where(x => x.Equals("hello")).Count();
}

private static readonly List<string> s_wordsField = new() { "hello", "world", "!" };
```

**Diagnostic Id**: [IDE0120](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0120)
